packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -352,15 +351,18 @@ export class FetchInstrumentation extends InstrumentationBase<
   private _prepareSpanData(spanUrl: string): SpanData {
     const startTime = core.hrTime();
     const entries: PerformanceResourceTiming[] = [];
-
     if (typeof window.PerformanceObserver === 'undefined') {
       return { entries, startTime, spanUrl };
     }
 
     const observer: PerformanceObserver = new PerformanceObserver(list => {
-      const entries = list.getEntries() as PerformanceResourceTiming[];
-      entries.forEach(entry => {
-        if (entry.initiatorType === 'fetch' && entry.name === spanUrl) {
+      const perfObsEntries = list.getEntries() as PerformanceResourceTiming[];
```

Comment:
Feels like we should have a separate PR to turn on `no-shadow` for eslint.  A quick trial of it shows 101 instances in this repo.

Reply:
If you think that would avoid more bugs, could you open an issue so someone can handle it at some point ? (i could have opened it but i don't really understand whats shadowing exactly)